query_server/query/src/metadata/usage_schema_provider/mod.rs

```diff
@@ -65,12 +64,15 @@ impl UsageSchemaProvider {
         USAGE_SCHEMA
     }
 
-    pub fn table(&self, session: &SessionCtx, name: &str) -> Result<Arc<dyn TableProvider>> {
-        let usage_schema_table = self
-            .table_factories
-            .get(name)
-            .ok_or_else(|| MetaError::TableNotFound { table: name.into() })?;
-        usage_schema_table.create(session, &self.default_table_provider)
+    pub fn table(
+        &self,
+        session: &SessionCtx,
+        name: &str,
+    ) -> Result<Option<Arc<dyn TableProvider>>> {
+        match self.table_factories.get(name) {
+            Some(f) => Ok(Some(f.create(session, &self.default_table_provider)?)),
+            None => Ok(None),
+        }
```

Comment:
what is the purpose here

Reply:
get table from usage_schema, if it doesn't exist return None